src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -305,6 +299,21 @@ private static function normalizeBody($body)
         return $body;
     }
 
+    private static function generatorToCallable(\Generator $body): \Closure {
```

Comment:
Not sure about the method: since this is a trait, it'd now be part of the BC promise, even if it's private.

Reply:
good catch. Reverted.

Move `is_string` before test on iterable (early return is more readable)